torch/linalg/__init__.py

```diff
@@ -713,16 +760,19 @@
 """)
 
 matrix_power = _add_docstr(_linalg.linalg_matrix_power, r"""
-matrix_power(input, n, *, out=None) -> Tensor
+matrix_power(A, n, *, out=None) -> Tensor
 
-Computes the :attr:`n`-th power of a matrix for an integer :attr:`n`. Supports batched inputs.
+Computes the n-th power of a square matrix for an integer n.
+
+Supports inputs of float, double, cfloat and cdouble dtypes.
+Also supports batched inputs, and, if the input is batched, the output is batched with the same dimensions.
 
 If :attr:`n` `= 0`, it returns the identity matrix (or batch) of the same shape
```

Comment:
This is mixing matrix and batch language but to be consistent with the new style I think it should just act as if "A" is a matrix?